### PR TITLE
Optimize Dockerfile for faster builds and smaller image size

### PR DESCRIPTION
**Pull request by Jannik-Schroeder**
This commit optimizes the Dockerfile for faster builds and smaller image size, as well as integrates Docker layer caching into the build and push workflow.

- Leverage Docker cache by copying package.json and yarn.lock first and installing dependencies before copying the rest of the application files.
- Combine related commands to reduce the number of layers in the image.
- Selectively copy only necessary files and directories at each stage to keep the image size smaller.

Note: The effectiveness of these optimizations is enhanced by the Docker layer caching mechanism introduced in PR #18. This caching setup will significantly speed up the build process for subsequent runs, reducing overall build time and improving efficiency.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,11 +3,15 @@ FROM node:alpine AS base
 
 WORKDIR /app
 
-COPY . .
-
-ARG NEXT_PUBLIC_API_URI=PLUNK_API_URI
```
Comment (driaug) [inline Dockerfile:8]: This line is needed. Next.js builds the applications and hard-writes public env variables into the bundled code.

We set this as a placeholder value and replace it on startup with the actual env variable.

**Review by driaug — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (Jannik-Schroeder) [reply Dockerfile:8]: Should be fixed now.

**Comment by driaug** [comment]
Let me quickly verify that this still builds correctly and get back to you!

Since we run a monorepo setup, I am quite sure that yarn also creates `node_modules` in the respective package folders which may not work if they do not exist yet.

**Comment by Jannik-Schroeder** [comment]
You're right, that will be a problem, but that is not to hard to change.

**Pull request closed without merging**